### PR TITLE
[Bug](top-n) do not update topn filter when sort node and scan node are not in the…

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
… same backend

## Proposed changes
do not update topn filter when sort node and scan node are not in the same backend
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/runtime/runtime_predicate.h [hunk 1]
```diff
@@ -46,13 +46,19 @@ class RuntimePredicate {
...
     Status init(PrimitiveType type, bool nulls_first, bool is_asc, const std::string& col_name);
 
-    bool inited() {
+    bool inited() const {
         std::unique_lock<std::shared_mutex> wlock(_rwlock);
         return _inited;
     }
 
+    bool need_update() const {
+        std::unique_lock<std::shared_mutex> wlock(_rwlock);
```
Comment (yiguolei) [inline be/src/runtime/runtime_predicate.h:55]: why name wlock? actuall it is a readlock

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**